src/util/child_common.c

```diff
@@ -822,9 +822,9 @@ errno_t child_debug_init(const char *logfile, int *debug_fd)
 
         *debug_fd = fileno(debug_filep);
         if (*debug_fd == -1) {
+            ret = errno;
             DEBUG(SSSDBG_FATAL_FAILURE,
                   "fileno failed [%d][%s]\n", errno, strerror(errno));
```

Comment:
Please, update also the `DEBUG` statement. Use `ret` here instead of `errno`.

Reply:
Thank you for the review.
Updated DEBUG statement to use ret instead of errno.